packages/next-swc/crates/next-core/src/next_app/mod.rs

```diff
@@ -282,6 +283,19 @@ impl Display for PathSegment {
 )]
 pub struct AppPath(pub Vec<PathSegment>);
 
+impl AppPath {
+    pub fn is_dynamic(&self) -> bool {
+        self.iter().any(|segment| {
+            matches!(
+                (segment,),
+                (PathSegment::Dynamic(_)
+                    | PathSegment::CatchAll(_)
+                    | PathSegment::OptionalCatchAll(_),)
```

Comment:
```suggestion
                segment,
                PathSegment::Dynamic(_)
                    | PathSegment::CatchAll(_)
                    | PathSegment::OptionalCatchAll(_),
```

Reply:
I wish, unfortunately the `matches!` macro doesn't accept that

The `|` makes rust think the pattern ends or something?